extensions-contrib/spectator-histogram/src/main/java/org/apache/druid/spectator/histogram/SpectatorHistogramAggregatorFactory.java

```diff
@@ -0,0 +1,373 @@
+  {
+    return new SpectatorHistogramAggregatorFactory(name, name);
+  }
+
+  @Override
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
+  {
+    if (other.getName().equals(this.getName()) && this.getClass() == other.getClass()) {
+      return getCombiningFactory();
+    } else {
+      throw new AggregatorFactoryNotMergeableException(this, other);
+    }
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Collections.singletonList(
+        new SpectatorHistogramAggregatorFactory(
+            fieldName,
+            fieldName
+        )
+    );
+  }
+
+  @Override
+  public Object deserialize(Object serializedHistogram)
+  {
+    return SpectatorHistogram.deserialize(serializedHistogram);
+  }
+
+  @Nullable
+  @Override
+  public Object finalizeComputation(@Nullable Object object)
+  {
+    return object;
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public String getFieldName()
+  {
+    return fieldName;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Collections.singletonList(fieldName);
+  }
+
+  @Override
+  public String getComplexTypeName()
+  {
+    return TYPE_NAME;
+  }
+
+  @Override
+  public ValueType getType()
+  {
+    return ValueType.COMPLEX;
+  }
+
+  @Override
+  public ValueType getFinalizedType()
+  {
+    return ValueType.COMPLEX;
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    return SpectatorHistogram.getMaxIntermdiateHistogramSize();
+  }
+
+  @Override
+  public AggregateCombiner makeAggregateCombiner()
+  {
+    return new ObjectAggregateCombiner<SpectatorHistogram>()
+    {
+      private SpectatorHistogram combined = null;
+
+      @Override
+      public void reset(final ColumnValueSelector selector)
+      {
+        combined = null;
+        fold(selector);
+      }
+
+      @Override
+      public void fold(final ColumnValueSelector selector)
+      {
+        SpectatorHistogram other = (SpectatorHistogram) selector.getObject();
+        if (other == null) {
+          return;
+        }
+        if (combined == null) {
+          combined = new SpectatorHistogram();
+        }
+        combined.merge(other);
+      }
+
+      @Nullable
+      @Override
+      public SpectatorHistogram getObject()
+      {
+        return combined;
+      }
+
+      @Override
+      public Class<SpectatorHistogram> classOfObject()
+      {
+        return SpectatorHistogram.class;
+      }
+    };
+  }
+
+  @Override
+  public boolean equals(final Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || !getClass().equals(o.getClass())) {
+      return false;
+    }
+    final SpectatorHistogramAggregatorFactory that = (SpectatorHistogramAggregatorFactory) o;
+
+    //TODO: samarth should we check for equality of contents in count arrays?
```

Comment:
Also to be resolved

Reply:
This was left over from an earlier implementation and no longer relevant.